pkg/resourcesets/collector.go

```diff
@@ -249,156 +229,79 @@ func (h *ResourceHandler) filterByLabel(ctx context.Context, dr dynamic.Resource
 		// no filters for names of the resource, return all objects obtained from the list call
 		return resourceObjectsList, nil
 	}
 
-	filteredByNameMap := make(map[string]bool)
+	// Map to store included names for fast lookup
+	includeNameMap := make(map[string]bool)
+	for _, name := range filter.ResourceNames {
+		includeNameMap[name] = true
+	}
 
-	// filter out using ResourceNameRegexp
+	var includeRegex *regexp.Regexp
 	if filter.ResourceNameRegexp != "" {
-		logrus.Debugf("Using ResourceNameRegexp [%s] to filter resource names", filter.ResourceNameRegexp)
-		for _, resObj := range resourceObjectsList {
-			if filter.ResourceNameRegexp != "." {
-				metadata := resObj.Object["metadata"].(map[string]interface{})
-				name := metadata["name"].(string)
-				nameMatched, err := regexp.MatchString(filter.ResourceNameRegexp, name)
-				if err != nil {
-					return filteredByName, err
-				}
-				if !nameMatched {
-					logrus.Debugf("Skipping [%s] because it did not match ResourceNameRegexp [%s]", name, filter.ResourceNameRegexp)
-					continue
-				}
-			}
-			filteredByName = append(filteredByName, resObj)
-			key := resObj.Object["filterUid"].(string)
-			filteredByNameMap[key] = true
-		}
+		includeRegex = regexp.MustCompile(filter.ResourceNameRegexp)
 	}
-
-	// filter out using ExcludeResourceNameRegexp
+	var excludeRegex *regexp.Regexp
 	if filter.ExcludeResourceNameRegexp != "" {
-		if filter.ResourceNameRegexp == "" {
-			filteredByName = resourceObjectsList
-		}
-		var newFilteredByName []unstructured.Unstructured
-		for _, resObj := range filteredByName {
-			metadata := resObj.Object["metadata"].(map[string]interface{})
-			name := metadata["name"].(string)
-			nameMatched, err := regexp.MatchString(filter.ExcludeResourceNameRegexp, name)
-			if err != nil {
-				return filteredByName, err
-			}
-			if nameMatched {
-				logrus.Debugf("Skipping [%s] because it did match ExcludeResourceNameRegexp [%s]", name, filter.ExcludeResourceNameRegexp)
-				key := resObj.Object["filterUid"].(string)
-				filteredByNameMap[key] = false
-				continue
-			}
-			newFilteredByName = append(newFilteredByName, resObj)
-			key := resObj.Object["filterUid"].(string)
-			filteredByNameMap[key] = true
-		}
-		filteredByName = newFilteredByName
-	}
-	// filter by names as fieldSelector:
-	if len(filter.ResourceNames) > 0 {
-		logrus.Debugf("Using ResourceNames [%s] to filter resource names", strings.Join(filter.ResourceNames, ","))
-		allowedNames := make(map[string]bool)
-		for _, name := range filter.ResourceNames {
-			allowedNames[name] = true
-		}
-		for _, resObj := range resourceObjectsList {
-			metadata := resObj.Object["metadata"].(map[string]interface{})
-			name := metadata["name"].(string)
-			if allowedNames[name] {
-				filteredByResourceNames = append(filteredByResourceNames, resObj)
-			}
-		}
+		excludeRegex = regexp.MustCompile(filter.ExcludeResourceNameRegexp)
```

Comment:
same consideration as above

Reply:
We don't have a preprocess pass to verify the regexes, but they're embedded in the chart, so it's unlikely that users will change or add to them. However this should use `regex.Compile` and return a non-nil error object through the normal channels so it can be surfaced gracefully (and if there is an invalid regex, it should be caught during integration testing, because running a backup will try to compile every submitted regex.

Reply:
ok cool